content/docs/guides/driver-design.md

```diff
@@ -95,10 +99,13 @@ func (d *Device) writeRegister(addr, newValue uint8) error {
     return d.spi.Tx(d.bufWrite[:], d.bufRead[:])
 }
 ```
+| Note on Go's array type in TinyGo |
+|------|
+| Note the use of static arrays types like `[2]byte`. "Arrays", as they are called in Go, if small enough will always be stack allocated. TinyGo currently has a **Max Stack Object Size** configuration which runs around the 256 byte size. That said, this fact is not relevant in this case since the array is attached to the `Device` struct which almost certainly is heap allocated. Most drivers, if not all, initialize and use a pointer type (`*Device`). If the `Device` is heap allocated then the arrays it contains will also be heap allocated- and this is OK! Remember, we are trying to constrain our allocations, not completely eliminate heap allocations. We win if during the lifetime of the program there are **ZERO** heap allocations. Allowing heap allocations at initialization is a good practice and actually encouraged in NASA's coding standards for extraplanetary missions. |
```

Comment:
Link to NASA coding standards? Can be educational!

Reply:
Will add it
https://en.m.wikipedia.org/wiki/The_Power_of_10:_Rules_for_Developing_Safety-Critical_Code